src/app/item-popup/ItemStat.tsx

```diff
@@ -176,3 +202,29 @@ function getModdedStatValue(item: DimItem, stat: DimStat) {
 export function isD1Stat(item: DimItem, _stat: DimStat): _stat is D1Stat {
   return item.isDestiny1();
 }
+
+function breakDownTotalValue(statValue: number, item: D2Item) {
+  let baseTotalValue = statValue;
+  let totalModsValue = 0;
+  let totalMasterworkValue = 0;
+
+  if (item.sockets && item.sockets.sockets) {
+    for (const socket of item.sockets.sockets) {
+      if (socket.plug && socket.plug.stats) {
+        for (const statHash of armorStats) {
+          const plugStatValue = socket.plug.stats[statHash];
+          if (plugStatValue) {
+            baseTotalValue -= plugStatValue;
+            if (socket.plug.isMasterwork) {
+              totalMasterworkValue += plugStatValue;
```

Comment:
Ah! I was just removing `isMasterwork` because nobody was using it. TBH I really don't know how masterworked Armor2.0 works. Maybe @sundevour has one?

Reply:
https://github.com/sundevour/sample-d2-api-responses/blob/master/stubs/BrayTech_Combat_Vest-masterwork-Y3-DimItem.json

![image](https://user-images.githubusercontent.com/31990469/67155482-6e07e100-f2c5-11e9-8267-da707496ba57.png)

Reply:
Interesting, so it isn't shown as a mod, just as a full energy bar.

Reply:
those +2s aren't anywhere in the item info except its livestats?

Reply:
I personally think the best way to distinguish masterwork armour would be this
'''
"energy": {
    "energyTypeHash": 4069572561,
    "energyType": 3,
    "energyCapacity": 10,
    "energyUsed": 10,
    "energyUnused": 0
  },
'''
By checking that the top level energy of the armour has 10 in energyUsed we can be sure of it. The manually move the 12 points around, as I don't see anything specifically useable otherwise.

The only plug I can see that refers to masterworks is this
'''
"plug": {
              "insertionRules": [],
              "plugCategoryIdentifier": "v460.plugs.armor.masterworks.stat.resistance_4",
              "plugCategoryHash": 400813045,
              "onActionRecreateSelf": false,
              "actionRewardSiteHash": 0,
              "actionRewardItemOverrideHash": 0,
              "insertionMaterialRequirementHash": 3588361558,
              "previewItemOverrideHash": 0,
              "enabledMaterialRequirementHash": 0,
              "enabledRules": [],
              "uiPlugLabel": "masterwork",
              "plugStyle": 1,
              "plugAvailability": 2,
              "alternateUiPlugLabel": "",
              "alternatePlugStyle": 0,
              "isDummyPlug": false,
              "energyCapacity": {
                "capacityValue": 10,
                "energyTypeHash": 4069572561,
                "energyType": 3
              }
},
'''

Whats the thoughts of doing it that way?

Reply:
Yeah, that looks like it - it's telling you in `investmentStats` what the value of the item being masterworked is.

Reply:
Cool I will update it to use that after work.

Reply:
I took another look BTW - there is a mod there that represents masterwork tier, we just skip displaying it in favor of our custom energy display. So you should be able to find it and use its stats.

Reply:
Ok cool, I have updated this. 
Using the plug on line 1417 I have used the stat block in it, specifically 
```
"stats": {
    "16120457": 10,
    "144602215": 2,
    "392767087": 2,
    "1735777505": 2,
    "1943323491": 2,
    "2996146975": 2,
    "4244567218": 2
}
```
I am looking for the 16120457 statHash with a value of 10 and assuming that is the energy value for the masterwork. 

I had a look at some data for my hunter and warlock but it appears you have a similar but ultimately different plug for unmasterworked armour. Mine didn't even have the statHashes for armour in it and had a single statHash of 3625423501 with the level of masterworking you had done as the value. I'll attach a screenshot for reference.
![image](https://user-images.githubusercontent.com/7344652/67186613-25226a80-f434-11e9-9e62-417567d18054.png)

Let me know thought's if this is satisfactory or not and I can modify as needed. This system for mods and masterworks is a little more complex than I thought it would be and I am seem to discover something new and interesting everytime I dig into it still.

Reply:
Nice! You can filter the stats on the mods to the `armorStats` to exclude things like energy.